### PR TITLE
feat: add js formatter tool

### PR DESCRIPTION
**Pull request by Charlielancaster**
This PR is to address issue #73 

For testing, check it out at https://js-formatter.web.app

## Before merging

- [x] add the completed tool as a subdomain in Google hosting panel. We want to configure entries for both `tool-name.holistic-toolbox.com` and `www.tool-name.holistic-toolbox.com`.
- [x] in the firebase console, configure the new site to use these new domains
- [x] Finish documentation about the tool and how to use/install it

image

image

### CODE REVIEW

File: tools/js-formatter/src/JsFormatter.vue [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+					beautify: this.whitespace === 'spaces',
+					comments: true,
+					quote_style: 1
+				}
+			};
+			try {
+				const { code, error } = uglify.minify(this.jsString, options);
+				if (error) throw new Error(error);
+				this.jsString = code;
+				this.formatted = true;
```
Comment (michaelfitzhavey) [inline tools/js-formatter/src/JsFormatter.vue]:
```suggestion
				this.formatted = true;
				this.errorMessage =  null;
```

Comment (andrewlew1s) [reply tools/js-formatter/src/JsFormatter.vue]: this is a great suggestion!

Comment (michaelfitzhavey) [reply tools/js-formatter/src/JsFormatter.vue]: ^ @Charlielancaster this should ensure that if JS is successfully formatted, then historical error messages will be cleared.

File: tools/js-formatter/README.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# JS Formatter
+
+This tool validates and formats JavaScript using [`UglifyJS`](https://www.npmjs.com/package/uglifyjs-browser)
```
Comment (michaelfitzhavey) [inline tools/js-formatter/README.md]: Let's replace the top sentence and links with the new standard pattern

```suggestion
This is a tool to format JS code.

## Factsheet
| **Category**            | **Value**                                  |
|-------------------------|--------------------------------------------|
| **Project Type**        | Tool: Web App                              |
| **Live URL**            | https://js-formatter.holistic-toolbox.com  |
| **Firebase Site Name**  | js-formatter                               |
| **Unique Dependencies** | vue-clipboard, uglifyjs-browser            |
```

File: tools/js-formatter/README.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+	```
+	npm run build
+	```
+2. Deploy the project
+	```
+	firebase deploy
+	```
+
+## Road Map
+- Allow the user to set more formatting options, possibly provide a small description against each option
```
Comment (michaelfitzhavey) [inline tools/js-formatter/README.md]:
```suggestion
- Allow the user to set more formatting options, possibly provide a small description against each option
- Support ES6 JS, such as `const a = 'test';`
```

File: tools/js-formatter/src/JsFormatter.vue (hunk 1 above)
Comment (Charlielancaster) [reply tools/js-formatter/src/JsFormatter.vue]: This change was done in commit 3c3c600

File: tools/js-formatter/README.md (hunk 3 above)
Comment (Charlielancaster) [reply tools/js-formatter/README.md]: This change was done in commit 7eaf19f

File: tools/js-formatter/README.md (hunk 2 above)
Comment (Charlielancaster) [reply tools/js-formatter/README.md]: This change was done in commit 7eaf19f

**Pull request merged**